core/src/main/kotlin/model/SourceSetData.kt

```diff
@@ -6,22 +6,33 @@ import org.jetbrains.dokka.plugability.DokkaContext
 
 data class SourceSetData(
     val moduleName: String,
-    val sourceSetName: String,
+    val sourceSetID: String,
+    val displayName: String,
     val platform: Platform,
     val sourceRoots: List<DokkaConfiguration.SourceRoot> = emptyList(),
-    val dependentSourceSets: List<String>  = emptyList()
+    val dependentSourceSets: List<String> = emptyList()
 )
 
 class SourceSetCache {
     private val sourceSets = HashMap<String, SourceSetData>()
 
     val allSourceSets: List<SourceSetData>
-    get() = sourceSets.values.toList()
+        get() = sourceSets.values.toList()
 
     fun getSourceSet(pass: DokkaConfiguration.PassConfiguration) =
-        sourceSets.getOrPut("${pass.moduleName}/${pass.sourceSetName}",
-            { SourceSetData(pass.moduleName, pass.sourceSetName, pass.analysisPlatform, pass.sourceRoots, pass.dependentSourceSets) }
+        sourceSets.getOrPut("${pass.moduleName}/${pass.sourceSetID}",
```

Comment:
> Maybe we should give the string "moduleName/sourceSetID" a formal name, since it is used multiple times? 
Sure, what do you suggest? 

> Also, can't we enforce a true unique sourceSetID? In that case, this would be simpler, because moduleName would not be required.
I'm not quite sure how to do it right. We could theoretically generate a unique string, but then we would have to do some translation from user input for dependent source sets to those internal ids, which basically boils down to the same thing (name used as an identifier), unless we force users to use other sourceSets reference and I'm not sure if that's a good idea